fakesnow/fakes.py

```diff
@@ -134,12 +134,16 @@ def execute(
             if os.environ.get("FAKESNOW_DEBUG") == "snowflake":
                 print(f"{command};{params=}" if params else f"{command};", file=sys.stderr)
 
+            command = self._inline_variables(command)
             command, params = self._rewrite_with_params(command, params)
             if self._conn.nop_regexes and any(re.match(p, command, re.IGNORECASE) for p in self._conn.nop_regexes):
                 transformed = transforms.SUCCESS_NOP
             else:
                 expression = parse_one(command, read="snowflake")
                 transformed = self._transform(expression)
+                if Variables.is_variable_modifier(transformed):
+                    self._conn.variables.update_variables(transformed)
+                    transformed = transforms.SUCCESS_NOP  # Nothing further to do if its a SET/UNSET operation.
```

Comment:
(optional) for consistency I think this could be handled like any other transform, eg: in `transforms.py` as the method
```python
def update_variables(
    expression: exp.Expression,
    variables: dict[str, Any],
) -> exp.Expression:
    # if is variable modifier, update variables dict and return transforms.SUCCESS_NOP
```